### ISSUE TITLE
Naming/FileName false positive for class/module name with more than one acronymns.

### ISSUE THREAD

**Issue by kpabi** [rubocop/rubocop#11956]
## Expected behavior
Creating a class MyFOOBAR in file my_foo_bar should not return a Naming/FileName

## Actual behavior
Creating such a file produced the error `Naming/FileName: my_foo.rb should define a class or module called MyFoo`
Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

## Steps to reproduce the problem
Add Gemfile in new directory and install rubocop
Add a Naming/FileName rule with two AllowedAcronyms: FOO, BAR
Create a class MyFOOBAR with filename my_foo_bar.rb
Run rubocop
It returns the message `Naming/FileName: my_foo_bar.rb should define a class or module called MyFooBar`
Doing the same with MyFOO should work with no offenses.
```

   

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

## RuboCop version
1.52.1

**Comment by camallen** [comment]
I'm hitting issues with multiple acronyms, using Rubocop version 1.75.5

Example is a file named `app/controllers/adp_run_us_session_controller.rb` that define a class 
``` ruby 
class ADPRunUSSessionController < ApplicationController
end
```

Rubocop config looks like 
```
Naming/FileName:
  ExpectMatchingDefinition: true
  AllowedAcronyms:
    - US
    - ADP
```

That results in the error 
> Offenses:
app/controllers/adp_run_us_session_controller.rb:1:1: C: Naming/FileName: adp_run_us_session_controller.rb should define a class or module called AdpRunUsSessionController.

Removing one of the acronym names resolves the issue
``` ruby 
class ADPRunSessionController < ApplicationController
end
```
> bundle exec rubocop --debug app/controllers/adp_run_session_controller.rb
1 file inspected, no offenses detected

``` ruby 
class RunUSSessionController < ApplicationController
end
```
> bundle exec rubocop --debug app/controllers/run_us_session_controller.rb
1 file inspected, no offenses detected

**Comment by bbatsov** [comment]
Reopened — still reproduces on master (1.88). With `AllowedAcronyms: [FOO, BAR]` and `ExpectMatchingDefinition: true`, `class MyFOOBAR` in `my_foo_bar.rb` is flagged as needing `MyFooBar`, while a single acronym (`MyFOO`) is accepted — back-to-back acronyms aren't handled.
